### DISCUSSION TITLE
hiding/removing stack trace on /api/

### DISCUSSION THREAD

**Discussion by vincenttan5** [tchiotludo/akhq#596] [Q&A]
I received a pen test report stating that our akhq /api/ is spitting out stacktrace which is a potential security risk. Is there a way that I can disable the stacktrace output?

/api/1/acls
/api/connect-cluster-configs/acls

spits out error as follows:

`{"message":"Internal Server Error: Error for Describe Acls config","_links":{"self":{"href":"/api/1/acls","templated":false}},"_embedded":{"stacktrace":{"message":"java.lang.RuntimeException: Error for Describe Acls config\n\tat org.akhq.utils.Logger.call(Logger.java:31)\n\tat org.akhq.modules.AbstractKafkaWrapper.describeAcls(AbstractKafkaWrapper.java:346)\n\tat org.akhq.modules.$KafkaWrapperRequestScopeDefinition$$exec15.invokeInternal(Unknown Source)\n\tat io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:146)\n\tat io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:73)\n\tat org.akhq.modules.$KafkaWrapperRequestScopeDefinition$Intercepted.describeAcls(Unknown Source)\n\tat org.akhq.repositories.AccessControlListRepository.findAll(AccessControlListRepository.java:25)\n\tat org.akhq.controllers.AclsController.list(AclsController.java:31)\n\tat org.akhq.controllers.$AclsControllerDefinition$$exec1.invokeInternal(Unknown Source)\n\tat io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:146)\n\tat io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:474)\n\tat io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:312)\n\tat io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118)\n\tat io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildResultEmitter$10(RoutingInBoundHandler.java:1363)\n\tat io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:35)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14918)\n\tat io.reactivex.Flowable.subscribe(Flowable.java:14865)\n\tat io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.run(FlowableSubscribeOn.java:82)\n\tat io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288)\n\tat io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253)\n\tat io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:79)\n\tat io.micrometer.core.instrument.Timer.lambda$wrap$0(Timer.java:148)\n\tat io.micronaut.scheduling.instrument.InvocationInstrumenterWrappedRunnable.run(InvocationInstrumenterWrappedRunnable.java:47)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)\n\tat java.base/java.lang.Thread.run(Unknown Source)\nCaused by: org.apache.kafka.common.config.ConfigException: Missing required configuration \"bootstrap.servers\" which has no default value.\n\tat org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:478)\n\tat org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:468)\n\tat org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:108)\n\tat org.apache.kafka.common.config.AbstractConfig.<init>(AbstractConfig.java:142)\n\tat org.apache.kafka.clients.admin.AdminClientConfig.<init>(AdminClientConfig.java:217)\n\tat org.apache.kafka.clients.admin.Admin.create(Admin.java:63)\n\tat org.apache.kafka.clients.admin.AdminClient.create(AdminClient.java:39)\n\tat org.akhq.modules.KafkaModule.getAdminClient(KafkaModule.java:100)\n\tat org.akhq.modules.AbstractKafkaWrapper.lambda$describeAcls$24(AbstractKafkaWrapper.java:349)\n\tat org.akhq.utils.Logger.call(Logger.java:20)\n\t... 25 more\n"}}}
`

**Comment by tchiotludo** [comment]
You can't for now, there is no options, but it will be simple with a configuration options. 
All is here : https://github.com/tchiotludo/akhq/blob/dev/src/main/java/org/akhq/controllers/ErrorController.java#L81
Just add a configuration properties and hide the stack trace if the configuration is available :+1: 
PR are welcome ;)
